### PR TITLE
Re-write Export by Date Report

### PR DESCRIPTION
**Pull request by konavivekramakrishna**
# Pull Requests Requirements

- [x] The PR title includes a brief description of the work done, including the Issue number if applicable.
- [x] The PR includes a video showing the changes for the work done.
- [x] The PR title follows conventional commit label standards.
- [x] The changes confirm to the OpenElis Global x3 Styleguide and design documentation.
- [ ] The changes include tests or are validated by existing tests.
- [x] I have read and agree to the Contributing Guidelines of this project.

## Screenshots

https://github.com/I-TECH-UW/OpenELIS-Global-2/assets/101407963/90675b48-6de9-4f05-a756-33bd4af4cb0f

## Related Issue

fixes : #895 

This pr adds new page for of study reports: Export by Date Report and moved encodeDate function into utils

### CODE REVIEW

File: frontend/src/components/reports/study/common/ReportByDateCSV.js [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+
+    window.open(url, "_blank");
+    setLoading(false);
+    setNotificationVisible(true);
+  };
+
+  const statusOptions =
+    props.report === "CIStudyExport"
+      ? [
+          {
```
Comment (mozzy11) [inline frontend/src/components/reports/study/common/ReportByDateCSV.js:121]: The Project codes are suppsoed to be pulled from the backend ie see [here](https://github.com/mozzy11/OpenELIS-Global-2/blob/develop_3x/src/main/java/org/openelisglobal/reports/action/implementation/ExportProjectByDate.java#L205-L219)

Youll build an end point of in the [DisplayListController](https://github.com/mozzy11/OpenELIS-Global-2/blob/develop_3x/src/main/java/org/openelisglobal/common/rest/DisplayListController.java) that returns the List<IdValuePair>  for the the Project code and  Project Id of these [projects](https://github.com/mozzy11/OpenELIS-Global-2/blob/develop_3x/src/main/java/org/openelisglobal/reports/action/implementation/ExportProjectByDate.java#L205-L219) .  see [example ](https://github.com/mozzy11/OpenELIS-Global-2/blob/develop_3x/src/main/java/org/openelisglobal/common/rest/DisplayListController.java#L376)

File: frontend/src/components/reports/study/common/ReportByDateCSV.js [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+            id: "28",
+            value: "Viral Load",
+          },
+          {
+            id: "21",
+            value: "Indeterminate",
+          },
+        ]
+      : [
+          {
```
Comment (mozzy11) [inline frontend/src/components/reports/study/common/ReportByDateCSV.js:143]: Same for thse Poject codes , this is the List of [Projects](https://github.com/mozzy11/OpenELIS-Global-2/blob/develop_3x/src/main/java/org/openelisglobal/reports/action/implementation/ExportTrendsByDate.java#L238-L263)

**Review by mozzy11 — CHANGES_REQUESTED** [review]
The project codes should be pulled from the resnt endpoind

**Comment by konavivekramakrishna** [comment]
@mozzy11 

can you please review

**Comment by konavivekramakrishna** [comment]
@mozzy11 

can you please review it


https://github.com/I-TECH-UW/OpenELIS-Global-2/assets/101407963/13b62712-82a0-416d-a52f-0301d8489739

**Review by mozzy11 — APPROVED** [review]

File: src/main/java/org/openelisglobal/common/rest/DisplayListController.java [hunk 3]
```diff
@@ -105,6 +109,38 @@ public class DisplayListController extends BaseRestController{
...
+
+        // Convert Project objects to IdValuePair and add to the list
+        for (Project project : projectList) {
+            projects.add(new IdValuePair(project.getId(), project.getProjectName()));
+        }
+
+        return projects;
+    }
+
+  	@GetMapping(value = "trendsprojects", produces = MediaType.APPLICATION_JSON_VALUE)
```
Comment (mozzy11) [inline src/main/java/org/openelisglobal/common/rest/DisplayListController.java:132]: improper code formating

File: src/main/java/org/openelisglobal/common/rest/DisplayListController.java [hunk 4]
```diff
@@ -105,6 +109,38 @@ public class DisplayListController extends BaseRestController{
 	protected static List<Integer> statusList;
 	protected static List<String> nfsTestIdList;
 
+	// Manually create an instance of ExportTrendsByDate
+    private ExportTrendsByDate exportTrendsByDate = new ExportTrendsByDate();
+
+	@Autowired
```
Comment (mozzy11) [inline src/main/java/org/openelisglobal/common/rest/DisplayListController.java:115]: improper formating

**Comment by mozzy11** [comment]
Thanks @konavivekramakrishna

**Pull request merged**